pkg/reconciler/taskrun/validate_resources.go

```diff
@@ -88,6 +93,16 @@ func validateParams(ctx context.Context, paramSpecs []v1beta1.ParamSpec, params
 	return nil
 }
 
+func correctAoStype(params []v1beta1.Param, neededParamsTypes map[string]v1beta1.ParamType) {
```

Comment:
My question of this is that is it ok if we put this function here? Since we are doing validation in these functions it may seem weird to add one function to do something different. If we want to move it to higher level then it may be put inside `reconcile`